src/sqlfluff/core/linter/linting_result.py

```diff
@@ -181,3 +184,32 @@ def tree(self) -> Optional[BaseSegment]:  # pragma: no cover
                 "path."
             )
         return self.paths[0].tree
+
+    def mark_failed_files_unfixable(self) -> Tuple[int, int]:
+        """Discard lint fixes for files with templating or parse errors.
+
+        Scan all LintedFiles:
+        - Files with none of these errors: No action
+        - Files with these errors (before filtering): Removes fixes from
+          SQLLintError objects, i.e. marks them "unfixable"
+
+        Also returns the total number of these errors before and after
+        "ignore/noqa" filters are applied.
```

Comment:
I wonder if the number of errors after "ignore/noqa" filters are applied should be part of this function?

Seems separate (and short) so could be pulled out.

Just concerned there's potential for confusion between the removal (aka "filtering") of TMP/PRS errors, and the filtering of `ignore`/`--noqa` errors.

Reply:
Done.